common/src/app/common/pages/changes_builder.cljc

```diff
@@ -286,37 +286,38 @@
          update-shape
          (fn [changes id]
            (let [old-obj (get objects id)
-                 new-obj (update-fn old-obj)
+                 new-obj (update-fn old-obj)]
+             (if (= old-obj new-obj)
+               changes
```

Comment:
Here the only change is adding this `(if`. The rest is the diff algorithm getting mad.